ooniprobe/View/TestResults/Footer/UploadFooterViewController.m

```diff
@@ -65,45 +65,46 @@ -(void)uploadResult{
 - (void)doUploadWithProgress {
     if (self.result == nil && self.measurement == nil && self.upload_all) {
         //upload ALL
-        SRKResultSet *notUploaded = [Measurement notUploadedMeasurements];
-        [self uploadMeasurements:notUploaded];
+        [self uploadMeasurements:[Measurement notUploadedMeasurements]];
     }
     else if (self.result != nil && self.measurement == nil && self.upload_all) {
         //upload all measurements of that result
-        SRKResultSet *notUploaded = [self.result notUploadedMeasurements];
-        [self uploadMeasurements:notUploaded];
+        [self uploadMeasurements:[self.result notUploadedMeasurements]];
     }
     else if (self.result != nil && self.measurement != nil && !self.upload_all) {
         //upload this measurement
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [MBProgressHUD HUDForView:self.navigationController.view].label.text = NSLocalizedFormatString(@"Modal.ResultsNotUploaded.Uploading", @"");
-        });
-        [self uploadSingleMeasurement:self.measurement];
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [MBProgressHUD HUDForView:self.navigationController.view].progress = 1;
-        });
+        [self uploadMeasurements:[NSArray arrayWithObject:self.measurement]];
     }
 }
 
--(void)uploadMeasurements:(SRKResultSet *)notUploaded{
+-(void)uploadMeasurements:(NSArray *)notUploaded {
+    [self uploadMeasurements:notUploaded startAt:0];
+}
+
+-(void)uploadMeasurements:(NSArray *)notUploaded startAt:(int)idx{
     if ([notUploaded count] == 0) return;
```

Comment:
Please, use `<= 0` here to also cover the unlikely but possible case where the index is negative.

Reply:
An NSArray count can't be negative by implementation. It return an unsigned NSInteger

Reply:
Gotcha